Embed block: Implement specific Embed preview

fluiddot opened the pull request:
`gutenberg` PR: https://github.com/WordPress/gutenberg/pull/33426
`WordPress-iOS` PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/16866
`WordPress-Android` PR: https://github.com/wordpress-mobile/WordPress-Android/pull/15051

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/3276

To test:
Follow the testing instructions from `gutenberg` PR: https://github.com/WordPress/gutenberg/pull/33426

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Add react-native-webview podspec

fluiddot committed:
Update third party podspecs

fluiddot committed:
Update FBReactNativeSpec podspec

fluiddot committed:
Update Gutenberg ref

fluiddot commented on third-party-podspecs/react-native-webview.podspec.json:
```diff
@@ -0,0 +1,24 @@
+    "osx": "10.13"
+  },
+  "source": {
+    "git": "https://github.com/react-native-webview/react-native-webview.git",
```
We should update the forked repo to point to the WordPress mobile one.

fluiddot replied:
This was addressed in [this commit](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3725/commits/9f45708eab965268f97a7a5d3465a23a7cb41562).

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Merge branch 'develop' into embed-block-preview

fluiddot committed:
Update third party podspecs

fluiddot committed:
Trigger PR checks

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update bundle

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update third party podspecs

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update bundle

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update bundle

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update bundle

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Merge branch 'develop' into embed-block-preview

# Conflicts:
#	bundle/ios/App.js
#	bundle/ios/App.js.map

fluiddot committed:
Merge branch 'develop' into embed-block-preview

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update bundle

fluiddot committed:
Bump Gutenberg ref

fluiddot committed:
Update third party podspecs

fluiddot committed:
Revert commit reference of FBReactNativeSpec podspec

fluiddot committed:
Bump Gutenberg ref

fluiddot committed:
Update bundle

fluiddot committed:
Merge branch 'develop' into embed-block-preview

# Conflicts:
#	bundle/ios/App.js
#	bundle/ios/App.js.map

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Merge branch 'develop' into embed-block-preview

# Conflicts:
#	bundle/ios/App.js
#	bundle/ios/App.js.map

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update bundle

ceyhun reviewed (approved):
Approved via https://github.com/WordPress/gutenberg/pull/33426

fluiddot committed:
Update Gutenberg ref

fluiddot committed:
Update Gutenberg ref with merge commit

fluiddot committed:
Update bundle

fluiddot commented:
Not sure why but the "Check Correctness" CI check is failing with error `cb() never called!`, I tried to restart it several times but keeps failing so I'm going to debug it just in case this PR is actually introducing the failure.

fluiddot commented:
> Not sure why but the "Check Correctness" CI check is failing with error `cb() never called!`, I tried to restart it several times but keeps failing so I'm going to debug it just in case this PR is actually introducing the failure.

I connected to the machine instance by SSH and I verified that executing the command `npm install --no-audit` succeeds, running it after removing all `node_modules` folders also succeeds, so I have no clue regarding what it's causing the failure in this case 🤔 .

fluiddot commented:
> > Not sure why but the "Check Correctness" CI check is failing with error `cb() never called!`, I tried to restart it several times but keeps failing so I'm going to debug it just in case this PR is actually introducing the failure.
> 
> I connected to the machine instance by SSH and I verified that executing the command `npm install --no-audit` succeeds, running it after removing all `node_modules` folders also succeeds, so I have no clue regarding what it's causing the failure in this case 🤔 .

Finally, I managed to reproduce it by removing the folder `$HOME/.npm`, however, it doesn't give any clue about why fails. I'm wondering if it's related to some kind of timeout 🤔 , I'll keep investigating and let you with my findings 🔍 .

fluiddot committed:
[TEST] Install Gutenberg deps with install command

fluiddot committed:
[TEST] Install dependencies in two steps

fluiddot committed:
[TEST] Use cache key fallback on npm install full

fluiddot committed:
Revert cache key fallback of npm install full

fluiddot commented:
I fixed the issue by adding a cache key fallback and re-running the jobs (it was done in [this commit](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3725/commits/ad46787d099a627aeeb4cf3fd65dccdc61fc03aa)), this way the NPM cache has been saved for the new checksum of the Gutenberg's `package-lock.json` file. After this, I reverted the changes and all CI checks have passed 🎊 .
